module/product/src/Domain/Entity/AbstractProduct.php

```diff
@@ -84,21 +84,22 @@ public function __construct(
     ) {
         Assert::allIsInstanceOf($categories, CategoryId::class);
 
-        $attributes = array_filter(
-            $attributes,
-            function ($value) {
-                return $value !== null;
-            }
-        );
-
         $this->apply(new ProductCreatedEvent(
             $id,
             $sku,
             $this->getType(),
             $templateId,
-            $categories,
-            $attributes
         ));
+
+        foreach ($categories as $categoryId) {
+            $this->addToCategory($categoryId);
+        }
+
+        foreach ($attributes as $code => $value) {
+            if ($value) {
```

Comment:
`array_filter` will cleanup `null` values from `$attributes`. That `if` will be erased